js/ui/date_box/ui.date_box.base.js

```diff
@@ -644,7 +647,7 @@ var DateBox = DropDownEditor.inherit({
             hasText = !!text && value !== null,
             isDate = !!value && typeUtils.isDate(value) && !isNaN(value.getTime()),
             isDateInRange = isDate && dateUtils.dateInRange(value, this.dateOption("min"), this.dateOption("max"), this.option("type")),
-            isValid = !hasText || !hasText && !value || isDateInRange,
+            isValid = !hasText && !value || isDateInRange,
```

Comment:
```javascript 
isValid = !hasText || !value || isDateInRange,
```
?

Reply:
No. If a value is null and text is incorrect the isValid option must have "false" value